app/src/org/commcare/logic/AndroidFormController.java

```diff
@@ -20,13 +21,17 @@ public class AndroidFormController extends FormController implements PendingCall
     private boolean wasPendingCalloutCancelled;
     private FormIndex formIndexToReturnTo = null;
     private boolean formCompleteAndSaved = false;
+    private InterruptedFormState restoredFormSession;
 
     private FormAnalyticsHelper formAnalyticsHelper;
 
-    public AndroidFormController(FormEntryController fec, boolean readOnly, FormIndex formIndex) {
+    public AndroidFormController(FormEntryController fec, boolean readOnly, InterruptedFormState savedFormSession) {
```

Comment:
can we mark `savedFormSession` and `restoredFormSession` as `@Nullable`

Reply:
[7218aba](https://github.com/dimagi/commcare-android/pull/2858/commits/7218aba874042f32b616a363889118bfc7bd2516)